runtimes/rust/src/runtime_api.rs

```diff
@@ -143,24 +143,29 @@ pub fn endpoints(runtime: &RustRuntime) -> impl IntoIterator<Item = (String, Api
         // This endpoint returns list of protobuf source files of the specified artifact,
         // otherwise it returns source files of Exonum itself.
         .endpoint("proto-sources", {
-            move |query: ProtoSourcesQuery| -> api::Result<Vec<ProtoSourceFile>> {
-                if let ProtoSourcesQuery::Artifact { name, version } = query {
-                    let artifact_id = ArtifactId::new(RuntimeIdentifier::Rust, name, version)
-                        .map_err(|e| {
+            move |query: ProtoSourcesQuery| {
```

Comment:
Can't this approach be used to minimize cloning?

```rust
|query: ProtoSourcesQuery| async move {
    // ...
}
```

(Similarly, in some other places.)

Reply:
This approach will work if the endpoint handler accepts `FnOnce` instead of` Fn`.